heapster-saw/src/Verifier/SAW/Heapster/Implication.hs

```diff
@@ -1927,6 +2011,35 @@ simplImplOut (SImpl_IntroLLVMFieldContents x _ fld) =
 simplImplOut (SImpl_DemoteLLVMFieldRW x fld) =
   distPerms1 x (ValPerm_Conj [Perm_LLVMField $
                               fld { llvmFieldRW = PExpr_Read }])
+simplImplOut (SImpl_SplitLLVMWordField x fp bv1 bv2 endianness) =
```

Comment:
Nowhere in this case do we actually do anything with this `bv2` that is passed in. Either we should verify that it is equal to the `bv2` which is the result of calling `bvSplit`, or we should get rid of this argument to `SImpl_SplitLLVMWordField` if it is not needed.

Reply:
Actually, the line

```
| Just (bv1, bv2) == bvSplit endianness knownNat bv ->
```

is testing that it is equal to the result of calling `bvSplit`. It is a little confusing, though, since if I had written `<-` in place of `==` then it would be a pattern match. I'm going to reorder the two sides of the equality so this isn't as confusing looking.